client/src/featureform/client.py

```diff
@@ -188,6 +192,135 @@ def close(self):
         """
         self.impl.close()
 
+    def add_trigger(self, trigger, resource):
+        """
+        Add a trigger to a resource
+
+        **Example:**
+        ```py title="definitions.py"
+        client.add_trigger("trigger_name", ("resource_name", "resource_variant, resource_type"))
+        ```
+
+        Args:
+            trigger(Union[str, TriggerResource]): The name of the trigger
+            resource(Union[tuple, FeatureColumnResource, TrainingSetVariant]): The name, variant and type of the resource
+        """
+        req = metadata_pb2.TriggerRequest()
+
+        trigger_req = metadata_pb2.Trigger()
+        if isinstance(trigger, str):
+            trigger_req.name = trigger
+        elif isinstance(trigger, TriggerResource):
+            trigger_req.name = trigger.name
+        else:
+            raise ValueError("Invalid trigger type")
```

Comment:
Might be helpful here to print the type passed to the function and then also provide the valid types so it's easier for the user to address the mistake.